src/instruction.rs

```diff
@@ -892,7 +898,15 @@ impl fmt::Display for Instruction {
             }) => {
                 write!(f, "PRAGMA {}", name)?;
                 if !arguments.is_empty() {
-                    write!(f, " {}", arguments.join(" "))?;
+                    let arguments = arguments
+                        .iter()
+                        .map(|arg| match arg {
+                            PragmaArgument::Identifier(i) => i.clone(),
+                            PragmaArgument::Integer(i) => format!("{}", i),
+                        })
+                        .collect::<Vec<_>>()
+                        .join(" ");
+                    write!(f, " {}", arguments)?;
```

Comment:
GitHub won't let me use the suggestion feature, so here's a code snippet that avoids extra string allocations:

```rust
for arg in &arguments {
    write!(f, " {}", arg)?;
}
```

Then elsewhere, add

```rust
impl fmt::Display for PragmaArgument {
    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
        match self {
            Self::Identifier(id) => write!(f, "{}", id),
            Self::Integer(i) => write!(f, "{}", i),
        }
    }
}
```

Reply:
(Could also just do the `match` inside of the `for` loop, but I prefer this since `PragmaArgument` is a `pub` type)